src/PIL/ImageGrab.py

```diff
@@ -149,29 +149,27 @@ def grabclipboard():
             session_type = None
 
         if shutil.which("wl-paste") and session_type in ("wayland", None):
-            output = subprocess.check_output(["wl-paste", "-l"]).decode()
-            mimetypes = output.splitlines()
-            if "image/png" in mimetypes:
-                mimetype = "image/png"
-            elif mimetypes:
-                mimetype = mimetypes[0]
-            else:
-                mimetype = None
-
-            args = ["wl-paste"]
-            if mimetype:
-                args.extend(["-t", mimetype])
+            args = ["wl-paste", "-t", "image"]
         elif shutil.which("xclip") and session_type in ("x11", None):
             args = ["xclip", "-selection", "clipboard", "-t", "image/png", "-o"]
         else:
             msg = "wl-paste or xclip is required for ImageGrab.grabclipboard() on Linux"
             raise NotImplementedError(msg)
 
         p = subprocess.run(args, capture_output=True)
-        err = p.stderr
-        if err:
-            msg = f"{args[0]} error: {err.strip().decode()}"
+        err = p.stderr.decode()
+        if p.returncode != 0:
+            allowed_errors = [
+                "Nothing is copied",  # wl-paste, when the clipboard is empty
+                "not available",  # wl-paste, when an image isn't available
+                "cannot convert",  # xclip, when an image isn't available
```

Comment:
When I have text in the clipboard, I get a "not available" message.

So that seems like the "not available" comment should mention that it also applies to xclip?
Could you explain how to replicate "cannot convert"?

<video src="https://github.com/python-pillow/Pillow/assets/3112309/28e24a09-4d22-41ad-bd28-49f061c37ada" />

Reply:
Mhh, I get this result. I tried copying from multiple applications, gnome terminal, firefox, kitty, and they all respond with the "cannot convert" error... 
[Screencast from 2024-01-27 09-06-37.webm](https://github.com/python-pillow/Pillow/assets/10795335/55da4603-beaa-4289-8998-55d74e7c12ef)

I will look into the xclip code to be sure

Reply:
After further inspection, it seems that the version debian uses (and thus ubuntu) is different from the upstream one, and does indeed error out with `fprintf(stderr, "Error: target %s not available\n", atom_name);` instead of `fprintf(stderr, " cannot convert %s selection to target '%s'\n", selection_name, atom_name);`. I will update the comment accordingly

Reply:
I see - https://github.com/astrand/xclip/blob/b372f73579d30f9ba998ffd0a73694e7abe2c313/xclip.c#L741-L749 vs https://sources.debian.org/src/xclip/0.13-2/xclip.c/?hl=468#L466-L468